docs/directory_structure.md

```diff
@@ -0,0 +1,59 @@
+# Repo directory structure
+
+## cmake
+CMake files for writing automated tests
+
+
+## docs
+Contains documentation on usage, design and technical details
+
+
+## examples
+Compartmentalized demo applications (not tested as part of the automated test
+suite)
+
+
+## external
+Third-party code. These are git subtrees of upstream codebases and each commit
+may only modify one subtree or the main repository.
+
+#### chromium
+The subset of chromium needed to build PartitionAlloc. See
+framework/partition-alloc for subtree's latest upstream commit and details on
+pulling upstream changes.
+
+#### nginx
+Nginx web server pulled from github.com/nginx/nginx mirror. Used with the RTMP
+module to demo compartmentalizing Nginx plugins.
+
+#### nginx-rtmp-module
+Nginx media streaming RTMP module from github.com/arut/nginx-rtmp-module
+
+
+## runtime 
+The compartmentalization runtime, including code for both the target process
+itself and for tracing the process.
+
+#### libia2
+The main compartmentalization library. Includes code for interposing libc and
+the API for compartmentalizing codebases where manual changes are required.
+
+#### partition-alloc
+A compartment-aware shim for PartitionAlloc.
+
+#### tracer
+TODO:
```

Comment:
The implementation of a separate process that tracks compartment memory ownership and enforces system call limitations to prevent compartment escape.

Reply:
Thanks!